.github/workflows/ci.yml

```diff
@@ -387,6 +387,31 @@ jobs:
         sudo ./ci/setup_grpc.sh
         ./ci/do_ci.sh cmake.exporter.otprotocol.test
 
+  cmake_modern_protobuf_grpc_test:
+    name: CMake test (with modern protobuf and grpc)
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
+    - name: setup
+      env:
+        PROTOBUF_VERSION: '23.2'
+        ABSEIL_CPP_VERSION: '20230125.3'
+        CXX_STANDARD: '17'
```

Comment:
Why do we need C++17 standard for build ?

Reply:
The new versions of abseil-cpp, protobuf and gRPC requires C++14, and this version of abseil-cpp seems has some compatibility problem(See https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/5099430537/jobs/9167125807 for details). So I keep the same setting with gRPC here.